eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreClientSpec.scala

```diff
@@ -1374,39 +1491,195 @@ class BitcoinCoreClientSpec extends TestKitBaseClass with BitcoindService with A
+        assert(signedTx.txIn.length == 2) // only one wallet input should have been added
+        wallet.publishTransaction(signedTx).pipeTo(sender.ref)
+        sender.expectMsg(signedTx.txid)
+        signedTx
+      }
+      localAnchorTx
+    }
+    // Both wallet utxos are now spent.
+    walletUtxos.foreach(utxo => assert(spliceTx.txIn.exists(_.outPoint.txid == utxo.txid) || anchorTx.txIn.exists(_.outPoint.txid == utxo.txid)))
+
+    // At that point, the mempool contains the following transactions:
+    //                   +------------------+     +------------------+     +-----------+
+    // funding_tx_1 ---->| remote_splice_tx |---->| remote_splice_tx |---->|           |
+    //                   +------------------+     +------------------+     | splice_tx |
+    //                                                  wallet_utxo_1 ---->|           |
+    //                                                                     +-----------+
+    //                   +-----------------+     +-----------+
+    // funding_tx_2 ---->| local_commit_tx |---->|           |
+    //                   +-----------------+     | anchor_tx |
+    //                        wallet_utxo_2 ---->|           |
+    //                                           +-----------+
+
+    // We now fill the mempool with large transactions to evict splice_tx and anchor_tx.
+    // We start with very large transactions and then use smaller transaction to completely fill the mempool.
+    parentTxs.take(12).foreach(tx => publishLargeTx(tx, 0, largeInputsCount, FeeratePerKw(5_000 sat)))
+    (0 until largeInputsCount).foreach(i => publishLargeTx(parentTxs.last, i, i + 1, FeeratePerKw(5_000 sat)))
+    Seq(spliceTx, anchorTx).foreach(walletTx => {
+      wallet.getMempoolTx(walletTx.txid).pipeTo(sender.ref)
+      assert(sender.expectMsgType[Failure].cause.getMessage.contains("Transaction not in mempool"))
+    })
+
+    // The wallet transactions have been evicted, but they must be kept inside the wallet and not be double-spent.
+    val txNotFunded = Transaction(2, Nil, Seq(TxOut(50_000 sat, Script.pay2wpkh(randomKey().publicKey))), 0)
+    wallet.fundTransaction(txNotFunded, FeeratePerKw(5000 sat), replaceable = true).pipeTo(sender.ref)
     assert(sender.expectMsgType[Failure].cause.getMessage.contains("Insufficient funds"))
-    // The transaction is kept in bitcoind's internal wallet.
-    wallet.rpcClient.invoke("gettransaction", tx.txid).map(json => Transaction.read((json \ "hex").extract[String])).pipeTo(sender.ref)
-    assert(sender.expectMsgType[Transaction].txid == tx.txid)
+    Seq(spliceTx, anchorTx).foreach(walletTx => {
+      wallet.rpcClient.invoke("gettransaction", walletTx.txid).map(json => Transaction.read((json \ "hex").extract[String])).pipeTo(sender.ref)
+      assert(sender.expectMsgType[Transaction].txid == walletTx.txid)
+    })
```

Comment:
🤓 
```suggestion
    Seq(spliceTx, anchorTx).foreach { walletTx =>
      wallet.rpcClient.invoke("gettransaction", walletTx.txid).pipeTo(sender.ref)
      val json = sender.expectMsgType[JValue]
      assert(!(json \ "trusted").extract[Boolean])
      assert(!(json \ "details" \\ "abandoned").extract[Boolean])
      assert(Transaction.read((json \ "hex").extract[String]).txid == walletTx.txid)
    }
```

Reply:
Good idea, done in https://github.com/ACINQ/eclair/pull/2817/commits/5ffbdb67e576ba54b3ee43be1217b9c1fba57a54